src/nlp/builder.py

```diff
@@ -644,41 +643,46 @@ def _build_single_dataset(self, split: Union[str, Split], run_post_process: bool
-                for checksums_dict in split_checksums_dicts.values()
-            )
-            if self.info.dataset_size is not None and self.info.download_size is not None:
-                self.info.size_in_bytes = (
-                    self.info.dataset_size + self.info.download_size + self.info.post_processing_size
+            post_processed = self._post_process(ds, resources_paths)
+            if post_processed is not None:
+                ds = post_processed
+                recorded_checksums = {}
+                for resource_name, resource_path in resources_paths.items():
+                    size_checksum = get_size_checksum_dict(resource_path)
+                    recorded_checksums[resource_name] = size_checksum
+                if verify_infos:
+                    if self.info.post_processed is None or self.info.post_processed.resources_checksums is None:
+                        expected_checksums = None
+                    else:
+                        expected_checksums = self.info.post_processed.resources_checksums.get(split)
+                    verify_checksums(expected_checksums, recorded_checksums, "post processing resources")
+                if self.info.post_processed is None:
+                    self.info.post_processed = PostProcessedInfo()
+                if self.info.post_processed.resources_checksums is None:
+                    self.info.post_processed.resources_checksums = {}
+                self.info.post_processed.resources_checksums[str(split)] = recorded_checksums
+                self.info.post_processing_size = sum(
+                    checksums_dict["num_bytes"]
+                    for split_checksums_dicts in self.info.post_processed.resources_checksums.values()
+                    for checksums_dict in split_checksums_dicts.values()
                 )
-            self._save_info()
+                if self.info.dataset_size is not None and self.info.download_size is not None:
+                    self.info.size_in_bytes = (
+                        self.info.dataset_size + self.info.download_size + self.info.post_processing_size
+                    )
+                self._save_info()
+                ds._info.post_processed = self.info.post_processed
+                ds._info.post_processing_size = self.info.post_processing_size
+                ds._info.size_in_bytes = self.info.size_in_bytes
+                if self.info.post_processed.features is not None:
+                    if self.info.post_processed.features.type != ds.features.type:
+                        raise ValueError(
+                            "Post-processed features info don't match the dataset:\nGot\n{}\nbut expected something like\n{}".format(
+                                self.info.post_processed.features, ds.features
+                            )
+                        )
+                    else:
+                        ds.info.features = self.info.post_processed.features
```

Comment:
When I added the unit test to check the good caching behavior of text datasets, I found that the `DatasetInfo` were not properly updated if there was a post processing, so I fixed that

Reply:
Good catch!